### PR TITLE
AE: fix passthrough for OSX sink

### PR DESCRIPTION
**Pull request by FernetMenta**
see title

cc @Memphiz

### CODE REVIEW

**Comment by Memphiz** [comment]
Wow. Thats intrusive. I am pretty sure by removing what was commented by "hack" in the rendercallback you removed a valid use case for some special osx audio hardware (where we need to reverse an float to int conversion which is done in the driver layer to pass through via 2.0 pcm - there is a forum thread somewhere i can dig up if you are interested in the circumstances). Also its a bit hard to follow what exactly you changed in the deviceenumeration class.

Are you open to an alternative way of fixing things with only touching 4 - 10 lines of code for beeing a bit more save against regressions for hardware we don't own? I would like to do a PR for the latter during the weekend.

**Comment by Memphiz** [comment]
Also your change now applies the channel map even on pass through configurations which looks like it will do harm too to me.

**Comment by FernetMenta** [comment]
the hack is not required anymore because I set the virtual format to be the same as the physical one. not touching the data is a requirement for passthrough.

**Comment by FernetMenta** [comment]
> Are you open to an alternative way of fixing things with only touching 4 - 10 lines

hmm, the old code is really in a bad shape with lots of dead code. the sink never transmitted raw data and query for data formats like AC3 is pretty useless. I don't intend to keep this including the hack

**Comment by Memphiz** [comment]
We need to find the user who had the setup which needed this hack to verify. Didn't find the thread in the forum yet. @jmarshallnz in case you are still around - any chance that you have a note somewhere which allows us to find that user who needed the reverse of the float 2 int16 hack?

**Comment by FernetMenta** [comment]
well, the entire enumeration process was incorrect. that may have required this hack. actually we only transmit pcm. for passthrough this means:
sample rate: 48000
channels: 2
bitsperchannel 16
physical format == virtual format

that's all

**Comment by jmarshallnz** [comment]
IIRC this was primarily for devices with optical connections that didn't report themselves as AC3/DTS capable, and would only report (or maybe only accept?) 32bit float. i.e. the conversion back to INT16 was done at the core audio driver level. We had to touch the passthrough'd audio as OSX drivers were going to touch it anyway.

Searching on the forums may help discover more? It's been way too long for specifics, and the commit msgs don't really give it away. You need to find someone with the hardware that repro's this. Possibly the fireaudio or whatever they were devices? I didn't have one; rather I just tested it by forcing it to use the hack path on my MB air via HDMI. i.e. select PCM 2 ch 32bit float in CA, set bistreaming bool, then tell AE we want RAW passthrough. Any receiver will then pick up that it's encoded and happily decode the IEC packing.

I have no idea whether or not the enumeration is/was correct or not - that's changed a bunch since I did anything.

**Comment by FernetMenta** [comment]
@jmarshallnz thanks for commenting, hope you are doing well

@Memphiz no need to search, it took that "hack" path on my system which was the reason I dug deeper. I heard some horrible noise for a second or two until I got normal audio.
I say it again: for iec packed passthrough we don't need to query any AC3/DTS capabilities. iec packed is transmitted as PCM and we only need to make sure that the driver doesn't touch it.

**Comment by FernetMenta** [comment]
@Memphiz updated according to your comments

jenkins build this please

**Comment by Memphiz** [comment]
thx - will see when i have time to test it

**Comment by Memphiz** [comment]
If found time to test it. It seems to work fine in general. But i have major asynchronities when using pass through and having 24p enabled. I doubt it is related to this very PR but just as a heads up. Beside that i really would like to have the dedicated ac3 streams chosen when scoring for ae raw format/iec passthrough (see my PR to your repo). Fine by me then...

**Comment by FernetMenta** [comment]
thanks.
jenkins build this please
